src/kv/lazy-store.test.ts

```diff
@@ -0,0 +1,639 @@
+  });
+  await baseStore.withWrite(async write => {
+    await write.put('inBase1', 'inBase1Value');
+    await write.put('inBase2', 'inBase2Value');
+    await write.commit();
+  });
+  await lazyStore.withRead(async read => {
+    expect(await read.get('inBase1')).to.equal('inCache1Value');
+    expect(await read.get('inBase2')).to.equal('inBase2Value');
+  });
+});
+
+test('values are reloaded from base store if evicted from cache', async () => {
+  const {baseStore, lazyStore} = createLazyStoreForTest();
+  await lazyStore.withRead(async read => {
+    expect(await read.get('inBase1')).to.be.undefined;
+  });
+  await baseStore.withWrite(async write => {
+    await write.put('inBase1', 'inBase1Value');
+    await write.put('inBase2', 'inBase2Value');
+    await write.commit();
+  });
+  await lazyStore.withRead(async read => {
+    expect(await read.get('inBase1')).to.equal('inBase1Value');
+    expect(await read.get('inBase2')).to.equal('inBase2Value');
+  });
+  await baseStore.withWrite(async write => {
+    await write.put('inBase1', 'inBase1V2Value');
+    await write.put('inBase2', 'inBase2V2Value');
+    await write.commit();
+  });
+  await lazyStore.withWrite(async write => {
+    expect(await write.get('inBase1')).to.equal('inBase1Value');
+    expect(await write.get('inBase2')).to.equal('inBase2Value');
+    await write.put('notPinned1', 'notPinned1Value');
+    await write.put('notPinned2', 'notPinned2Value');
+    await write.put('notPinned3', 'notPinned3Value');
+    // inBase1 and inBase2 evicted by above write of notPinned[1-2]
+    await write.commit();
+  });
+  await lazyStore.withRead(async read => {
+    // new values loaded from base store
+    expect(await read.get('inBase1')).to.equal('inBase1V2Value');
+    expect(await read.get('inBase2')).to.equal('inBase2V2Value');
+  });
+});
+
+test('values are reloaded from base store if deleted from cache', async () => {
+  const {baseStore, lazyStore} = createLazyStoreForTest();
+  await lazyStore.withRead(async read => {
+    expect(await read.get('inBase1')).to.be.undefined;
+  });
+  await baseStore.withWrite(async write => {
+    await write.put('inBase1', 'inBase1Value');
+    await write.put('inBase2', 'inBase2Value');
+    await write.commit();
+  });
+  await lazyStore.withRead(async read => {
+    expect(await read.get('inBase1')).to.equal('inBase1Value');
+    expect(await read.get('inBase2')).to.equal('inBase2Value');
+  });
+  await baseStore.withWrite(async write => {
+    await write.put('inBase1', 'inBase1V2Value');
+    await write.put('inBase2', 'inBase2V2Value');
+    await write.commit();
+  });
+  await lazyStore.withWrite(async write => {
+    expect(await write.get('inBase1')).to.equal('inBase1Value');
+    expect(await write.get('inBase2')).to.equal('inBase2Value');
+    await write.del('inBase1');
+    await write.del('inBase2');
+    await write.commit();
+  });
+  await lazyStore.withRead(async read => {
+    // new values loaded from base store
```

Comment:
Is this the right behavior?

I was uncertain about this earlier and still am.

Trying to think through potential problems:
- Data chunks should be fine because those are immutable
- GC
  - Meta (aka refs key) chunks are not written when the refs are `[]`. Should be fine since these never change.
  - RefCount chunks are deleted when the count gets to 0
    - ~~If the count goes to 0 in the memdag it is deleted there, but if read again we will get the old count from the perdag.~~
    - https://github.com/rocicorp/replicache/blob/main/src/dag/write.ts#L157
    - https://github.com/rocicorp/replicache/blob/main/src/dag/write.ts#L138-L139

## BUT

GC happens at commit time so I think there is no problem after all.